tests/tasks/snowflake/test_snowflake.py

```diff
@@ -45,3 +45,29 @@ def test_execute_error_must_pass_through(self, monkeypatch):
 
         with pytest.raises(sf.errors.DatabaseError, match="Invalid query"):
             task.run(query="SELECT * FROM foo")
+
+    def test_execute_fetchall(self, monkeypatch):
```

Comment:
For tests like this, a good practice is to add a docstring explaining the test -- in this case, you don't _actually_ care that `fetchall` is called, you care that the result is cloudpickleable.  The docstring ensures that someone who changes to another method call and this test fails, they know what they need to validate to safely delete / change the test.

Reply:
Can definitely add the docstring but just to clarify, this is really intended for the `fetchall` call because it means that the data was pulled from the Snowflake Query cursor and it would prevent future code edits from just returning the cursor.

I can't really test retrieving data from Snowflake and pickling that because I can't spin up a test database. In my tests though, `fetchall` returns a list.

Reply:
Right -- but if the API changes or someone uses a different method call here for whatever reason to something else that also doesn't return a cursor, they might delete this test when it inevitably fails, but the docstring will soft-enforce that the test is in place for pickleability, not because we have an opinion on using `fetchall`.